### PR TITLE
fix completion module path (#19351)(#19367)

### LINKED ISSUE #19351: Extend import autocomplete to `export from`
**TypeScript Version:**  2.5

**Code**

```ts
export * from './ModuleSpeci'; 
                
```

**Expected behavior:**

Intellisence suggestion for existing `ModuleSpecifier` module

**Actual behavior:**

No suggestions

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #19351 #19367

### CODE REVIEW

**Comment by Kingwl** [comment]
@mhegazy   
did i missing some syntax? 
review need

File: tests/cases/fourslash/completionForStringLiteralExport.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+// @typeRoots: my_typings
+
+// @Filename: test.ts
+
+//// export * from  "./[|some|]/*0*/
+//// export * from  "./sub/[|some|]/*1*/";
+//// export * from  "[|some-|]/*2*/";
+//// export * from  "../[||]/*3*/";
+//// export {} from  "./[|some|]/*4*/";
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/completionForStringLiteralExport.ts]: Do you need the range markers here? If it's not too much trouble, could you remove them?

**Comment by DanielRosenwasser** [comment]
I think the only thing I'm seeing missing here is dynamic imports expressions. I filed an issue if you want to tackle it here or in another PR: #19367

File: tests/cases/fourslash/completionForStringLiteralExport.ts (hunk 1 above)
Comment (Kingwl) [reply tests/cases/fourslash/completionForStringLiteralExport.ts]: sure😂

**Comment by Kingwl** [comment]
i'm going to take a look

**Comment by AngryPowman** [comment]
It's very helpful to me

**Comment by Kingwl** [comment]
@DanielRosenwasser seems ci is broken...

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -1390,6 +1390,27 @@ namespace ts {
         return !checkArgumentIsStringLiteral || arg.kind === SyntaxKind.StringLiteral || arg.kind === SyntaxKind.NoSubstitutionTemplateLiteral;
     }
 
+    /**
+     * Returns true if the node is a ImportCall with
+     * exactly one argument (of the form 'import("name")').
+     * This function does not test if the node is in a JavaScript file or not.
+     */
+    export function isDynamicImportCall(callExpression: Node): callExpression is CallExpression {
```
Comment (mhegazy) [inline src/compiler/utilities.ts]: consider:

```ts
export function isDynamicImportCall(node: Node): node is CallExpression {
    return node.kind === SyntaxKind.CallExpression &&
        (node as CallExpression).kind === SyntaxKind.ImportKeyword &&
        (node as CallExpression).args.length === 1;
}
```

Comment (mhegazy) [reply src/compiler/utilities.ts]: why not use `isImportCall` instead?

File: src/services/completions.ts [hunk 3]
```diff
@@ -217,11 +217,15 @@ namespace ts.Completions {
             // a['/*completion position*/']
             return getStringLiteralCompletionEntriesFromElementAccess(node.parent, typeChecker, compilerOptions.target, log);
         }
-        else if (node.parent.kind === SyntaxKind.ImportDeclaration || isExpressionOfExternalModuleImportEqualsDeclaration(node) || isRequireCall(node.parent, /*checkArgumentIsStringLiteral*/ false)) {
+        else if (node.parent.kind === SyntaxKind.ImportDeclaration || node.parent.kind === SyntaxKind.ExportDeclaration
+            || isRequireCall(node.parent, /*checkArgumentIsStringLiteral*/ false) || isDynamicImportCall(node.parent)
```
Comment (mhegazy) [inline src/services/completions.ts]: `|| isImportCall(node.parent)`

Comment (Kingwl) [reply src/services/completions.ts]: fxied

File: src/compiler/utilities.ts (hunk 2 above)
Comment (Kingwl) [reply src/compiler/utilities.ts]: fixed

**Pull request merged**